p2p/starknet/handlers.go

```diff
@@ -342,6 +475,45 @@ func (h *Handler) processIterationRequest(iteration *spec.Iteration, finMsg prot
 	}, nil
 }
 
+type iterationProcessorMulti = func(it blockDataAccessor) ([]proto.Message, error)
+
+func (h *Handler) processIterationRequestMulti(iteration *spec.Iteration, finMsg proto.Message,
+	getMsg iterationProcessorMulti,
+) (iter.Seq[proto.Message], error) {
+	it, err := h.newIterator(iteration)
+	if err != nil {
+		return nil, err
+	}
+
+	type yieldFunc = func(proto.Message) bool
+	return func(yield yieldFunc) {
+		// while iterator is valid
+		for it.Valid() {
+			// pass it to handler function (some might be interested in header, others in entire block)
+			messages, err := getMsg(it)
+			if err != nil {
+				h.log.Errorw("Failed to generate data", "blockNumber", it.BlockNumber(), "err", err)
+				break
+			}
+
+			for _, msg := range messages {
+				// push generated msg to caller
+				if !yield(msg) {
+					// if caller is not interested in remaining data (example: connection to a peer is closed) exit
+					// note that in this case we won't send finMsg
+					return
+				}
+			}
```

Comment:
If I am not mistaken, this is the only part which is different from `processIterationRequest`. Do you think we could pass this as a function and reduce the duplication?

Reply:
Having `multi` version simplify some of the handlers significantly so I think we can
just remove non multi call implementation and just assume that slice can be given from a handler. wdyt?

Reply:
Yeah sounds good to me. Happy with any approach which removes the duplication.